tests/integration-tests/conftest_utils.py

```diff
@@ -0,0 +1,169 @@
+        feature, test_id = item.nodeid.split("/", 1)
+        test_id = test_id.split("[", 1)[0]
+        dimensions = {}
+        for dimension in DIMENSIONS_MARKER_ARGS:
+            value = item.callspec.params.get(dimension)
+            if value:
+                dimensions[dimension + "s"] = [value]
+
+        if not dict_has_nested_key(failed_tests, ("test-suites", feature, test_id)):
+            dict_add_nested_key(failed_tests, [], ("test-suites", feature, test_id, "dimensions"))
+        if dimensions not in failed_tests["test-suites"][feature][test_id]["dimensions"]:
+            failed_tests["test-suites"][feature][test_id]["dimensions"].append(dimensions)
+            with open(out_file, "w", encoding="utf-8") as f:
+                yaml.dump(failed_tests, f)
+
+
+def extract_tested_component_from_filename(item: pytest.Item):
+    """Extract portion of test item's filename identifying the component it tests."""
+    test_location = os.path.splitext(os.path.basename(item.location[0]))[0]
+    return re.sub(r"test_|_test", "", test_location)
+
+
+def add_filename_markers(items: List[pytest.Item], config: pytest.Config):
+    """Add a marker based on the name of the file where the test case is defined."""
+    for item in items:
+        marker = extract_tested_component_from_filename(item)
+        # This dynamically registers markers in pytest so that warning for the usage of undefined markers are not
+        # displayed
+        config.addinivalue_line("markers", marker)
+        item.add_marker(marker)
+
+
+def runtest_hook_start_end_time(item: pytest.Item, when: str):
+    """Generator function to store start and end times for test phases."""
+    logging.info(f"Starting {when} for test {item.name}")
+    item.user_properties.append((f"start_time_{when}", datetime.now(timezone.utc)))
+    # execute all other hooks to obtain the call object
+    outcome: pluggy.Result = yield
+    item.user_properties.append((f"end_time_{when}", datetime.now(timezone.utc)))
+    call_list: List[pytest.CallInfo] = outcome.get_result()
+    logging.info(f"{when} list {call_list}")
+
+
+def publish_test_metrics(when: str, item: pytest.Item, rep: pytest.TestReport):
+    """
+    Publish test metrics specific to a given test execution.
+
+    Dimensions - feature, test name, region, os, instance type
+    Execution times - for each phase and total
+    Test Result - Pass/Fail
+    """
+    pub = MetricsPublisher(get_user_prop(item, "region"))
+    dimensions = [
+        {"Name": dimension, "Value": get_user_prop(item, dimension)}
+        for dimension in ["feature", "os", "instance", "region"]
+    ]
+    dimensions.append({"Name": "test_name", "Value": item.location[2]})
+    # Create a list of metrics
+    logging.info("create metrics")
+    metrics = create_phase_metrics(when, item, rep, dimensions)
+    logging.info("publish metrics")
+    pub.publish_metrics_to_cloudwatch("test_space_5", metrics)
```

Comment:
Why is the name space set to "test_space_5"? I think that a more descriptive name would be beneficial.

Reply:
oh thanks that was a local test space!  I will use an integ test name space